components/brave_wallet/browser/simple_hash_client.cc

```diff
@@ -644,4 +1229,39 @@ GURL SimpleHashClient::GetSimpleHashNftsByWalletUrl(
   return url;
 }
 
+GURL SimpleHashClient::GetNftsUrl(
+    mojom::CoinType coin,
+    const std::vector<mojom::NftIdentifierPtr>& nft_identifiers) {
+  if (nft_identifiers.empty()) {
+    return GURL();
+  }
+
+  std::string query_params;
+  size_t max_items =
+      std::min({nft_identifiers.size(), size_t(kSimpleHashMaxBatchSize)});
+  for (size_t i = 0; i < max_items; i++) {
+    std::optional<std::string> simple_hash_chain_id =
+        ChainIdToSimpleHashChainId(nft_identifiers[i]->chain_id);
+    if (simple_hash_chain_id) {
+      if (coin == mojom::CoinType::SOL) {
+        query_params +=
+            *simple_hash_chain_id + "." + nft_identifiers[i]->contract_address;
+      } else {
+        query_params += *simple_hash_chain_id + "." +
+                        nft_identifiers[i]->contract_address + "." +
+                        nft_identifiers[i]->token_id;
+      }
+      if (i < max_items -
+                  1) {  // Check to ensure we do not append a comma at the end
+        query_params += ",";
+      }
+    }
```

Comment:
It seems some callers rely on URL query having all provided nft_identifiers in it, but in case of wrong chain some of them would be skipped. Would that be safer to fail whole process when ChainIdToSimpleHashChainId fails?

Reply:
Agree. Updated and added a testcase.